crates/uv/tests/tree.rs

```diff
@@ -386,3 +432,115 @@ fn dev_dependencies() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn optional_dependencies() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        # ...
+        requires-python = ">=3.12"
+        dependencies = ["iniconfig", "flask[dotenv]"]
+
+        [project.optional-dependencies]
+        dev = ["anyio"]
+    "#,
+    )?;
+
+    uv_snapshot!(context.filters(), context.tree(), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    project v0.1.0
+    ├── flask[dotenv] v3.0.2
+    │   ├── blinker v1.7.0
+    │   ├── click v8.1.7
+    │   │   └── colorama v0.4.6
+    │   ├── itsdangerous v2.1.2
+    │   ├── jinja2 v3.1.3
+    │   │   └── markupsafe v2.1.5
+    │   ├── werkzeug v3.0.1
+    │   │   └── markupsafe v2.1.5
+    │   └── python-dotenv v1.0.1 (extra: dotenv)
+    ├── iniconfig v2.0.0
+    └── anyio v4.3.0 (extra: dev)
+        ├── idna v3.6
+        └── sniffio v1.3.1
+
+    ----- stderr -----
+    warning: `uv tree` is experimental and may change without warning
+    Resolved 14 packages in [TIME]
+    "###
+    );
+
+    // `uv tree` should update the lockfile
+    let lock = fs_err::read_to_string(context.temp_dir.join("uv.lock"))?;
+    assert!(!lock.is_empty());
+
+    Ok(())
+}
+
+#[test]
+fn optional_dependencies_inverted() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        # ...
+        requires-python = ">=3.12"
+        dependencies = ["iniconfig", "flask[dotenv]"]
+
+        [project.optional-dependencies]
+        dev = ["anyio"]
+    "#,
+    )?;
+
+    uv_snapshot!(context.filters(), context.tree().arg("--invert"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    blinker v1.7.0
+    └── flask v3.0.2
+        └── project[dotenv] v0.1.0
```

Comment:
The inverted form is kind of bad, the annotations are sort of on the wrong node, but it's honestly really hard to improve. I considered omitting them altogether.